matic/contracts/WildcardSteward_matic_v2.sol

```diff
@@ -1244,4 +1244,50 @@ contract WildcardSteward_matic_v1 is Initializable, BasicMetaTransaction {
 
         assetToken.transferFrom(_currentOwner, address(this), tokenId);
     }
+
+    function transferWithDeposit(
+        address to,
+        uint256 tokenId,
+        uint256 newPrice,
+        uint256 tokensDeposit
+    )
+        public
+        collectPatronageAndSettleBenefactor(tokenId)
+        collectPatronagePatron(to)
+        priceGreaterThanZero(newPrice)
+        youCurrentlyAreNotInDefault(msgSender())
+        youCurrentlyAreNotInDefault(to) // Maybe this condition is too strict? Maybe it should check foreclosure time after recieving the token rather?
+    {
+        address owner = assetToken.ownerOf(tokenId);
+        require(owner == msgSender(), "not owner");
+
+        // should be logically impossible for this to ever not be true.
+        assert(state[tokenId] == StewardState.Owned);
+
+        // timeLeftOfDeposit = deposit / rateOfDepositUse
+        //                   = deposit / (tokensYearlyPatronage / yearTimePatronagDenominator)
+        //                   = deposit / ((newPrice * tokenPatronageNumerator) / yearTimePatronagDenominator)
+        //                   = (deposit * yearTimePatronagDenominator) / (newPrice * tokenPatronageNumerator)
+        uint256 timeLeftOfDeposit =
+            tokensDeposit.mul(yearTimePatronagDenominator).div(
+                newPrice.mul(patronageNumerator[tokenId])
+            );
+        require(
+            timeLeftOfDeposit > 604800, /* seconds in a week */
+            "Deposit less than 1 week"
+        );
+
+        receiveErc20(tokensDeposit, msgSender());
+
+        // We won't charge users to use this function:
+        // _distributePurchaseProceeds(tokenId);
+
+        // Set the pecentage that wildcards will recieve when this token is purchased at 15%
+        serviceProviderPercentages[tokenId] = 150000;
+
+        deposit[to] = deposit[to].add(tokensDeposit);
```

Comment:
Looking closer now but are we subtracting deposit from the sender?

Reply:
nevermind I see its coming in the form of erc20 being taken from the sender, not from the senders deposit